cloudinit/distros/__init__.py

```diff
@@ -230,6 +236,16 @@ def _apply_network_from_network_config(self, netconfig, bring_up=True):
     def generate_fallback_config(self):
         return net.generate_fallback_config()
 
+    def _is_netplan(self) -> bool:
```

Comment:
Since most of this method is doing the same thing as the beginnning of `_write_network_state`, I think it might make more sense to fetch the renderer one time at the beginning of `apply_network_config` and pass it where it's needed.

Reply:
Done.